### PR TITLE
TEST-#15: Add get_history and apply_history tests

### LINKED ISSUE #15: Add testing for history functions
Also need to add fixes to apply_history

### LINKED ISSUE #16: Cell Edits through API don't always reflect in UI
Code to reproduce:
def create_df():
    return pd.DataFrame(
        {
            "A": 1.0,
            "Date": pd.Timestamp("20130102"),
            "C": pd.Series(1, index=list(range(4)), dtype="float32"),
            "D": np.array([3] * 4, dtype="int32"),
            "E": pd.Categorical(["test", "train", "foo", "bar"]),
            "F": ["foo", "bar", "buzz", "fox"],
        }
    )

df = create_df()
spreadsheet = SpreadsheetWidget(df=df)

spreadsheet._handle_view_msg_helper(
    {
        "row_index": 2,
        "column": "A",
        "unfiltered_index": 3,
        "value": 3,
        "type": "edit_cell",
    }
)

### LINKED ISSUE #18: Cell edits don't reflect across duplicate widget instances
### Environment

* Operating System: Mac OSX 10.15.7
* Python Version: 3.8.5
* How did you install modin-spreadsheet: pip
* Python packages: modin-spreadsheet 0.1.1

### Description of Issue

* What did you expect to happen?
If you create a spreadsheet widget and display in two different cells, then edit the cells in one of the widgets. The 2nd widget should update with the same value automatically. 

* What happened instead?
The 2nd widget doesn't update with the value unless you change the viewport and force a re-render.

### Reproduction Steps

1. Create a spreadsheet widget
2. Display the widget in two separate cells
3. Edit the cell value of one of the widgets

### What steps have you taken to resolve this already?
Added `update_table` function to end of `edit_cell` logic in PR#17 

### Anything else?
I don't see displaying the same widgets multiple times as a common use case, issue is not high priority.

### PR DESCRIPTION
**Pull request by valerieyip**
## What do these changes do?

- Fixes bugs with record transformation (resolves #16 , resolves #18 )
- Adds tests for get_history and apply_history

- [X] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [] passes `flake8 .`
- [X] passes `black --check .`
- [X] signed commit with `git commit -s` 
- [x] Resolves #15  
- [X] tests added and passing

### CODE REVIEW

**Review by devin-petersohn — CHANGES_REQUESTED** [review]
@valerieyip I think you accidentally added a lot of unnecessary files, would you be able to check?

File: modin_spreadsheet/grid.py [hunk 1]
```diff
@@ -1490,7 +1490,7 @@ def _handle_change_filter(self, content):
 
     def _handle_view_msg(self, widget, content, buffers=None):
         try:
-            self._view_msgs.append(content)
+            # self._view_msgs.append(content)
```
Comment (richardlin047) [inline modin_spreadsheet/grid.py:1493]: Get rid of comment

File: modin_spreadsheet/tests/test_grid.py [hunk 2]
```diff
@@ -753,3 +753,199 @@ def test_edit_cell():
...
+
+
+def test_get_history():
+    spreadsheet = SpreadsheetWidget(df=create_df())
+
+    last_history = spreadsheet.get_history()[-1]
+    expected_history = "unfiltered_df = df.copy()"
+    assert last_history == expected_history
+
+    # EDIT_CELL
```
Comment (richardlin047) [inline modin_spreadsheet/tests/test_grid.py:765]: Nit: Change to lowercase e.g. `# Edit cell`

File: modin_spreadsheet/tests/test_grid.py [hunk 3]
```diff
@@ -753,3 +753,199 @@ def test_edit_cell():
...
+            "value": "2013-01-10T00:00:00.000+00:00",
+            "type": "edit_cell",
+        },
+    )
+
+    last_history = spreadsheet.get_history()[-1]
+    expected_history = "# Edit cell\ndf.loc[(0, 'Date')]=Timestamp('2013-01-10 00:00:00')\nunfiltered_df.loc[(0, 'Date')]=Timestamp('2013-01-10 00:00:00')"
+    assert last_history == expected_history
+
+    # REMOVE_ROW
```
Comment (richardlin047) [inline modin_spreadsheet/tests/test_grid.py:794]: Same with the rest.

File: modin_spreadsheet/tests/test_grid.py [hunk 4]
```diff
@@ -753,3 +753,199 @@ def test_edit_cell():
...
+        {"type": "remove_row"},
+    )
+
+    last_history = spreadsheet.get_history()[-1]
+    expected_history = (
+        "# Remove rows\ndf.drop([], inplace=True)\nunfiltered_df.drop([], inplace=True)"
+    )
+    assert last_history == expected_history
+
+    # CHANGE_SELECTION
```
Comment (richardlin047) [inline modin_spreadsheet/tests/test_grid.py:805]: Add comment saying that you don't expect history to change. Same with Change viewport.

File: modin_spreadsheet/tests/test_grid.py [hunk 5]
```diff
@@ -753,3 +753,199 @@ def test_edit_cell():
...
+        {
+            "row_index": 2,
+            "column": "A",
+            "unfiltered_index": 3,
+            "value": 3,
+            "type": "edit_cell",
+        }
+    )
+
+    changed_df = spreadsheet.get_changed_df()
```
Comment (richardlin047) [inline modin_spreadsheet/tests/test_grid.py:948]: Add assert that df_copy doesn't equal changed df.

File: modin_spreadsheet/tests/test_grid.py [hunk 6]
```diff
@@ -753,3 +753,199 @@ def test_edit_cell():
...
+    # CLEAR_HISTORY
+    spreadsheet._handle_view_msg_helper({"type": "clear_history"})
+    history = spreadsheet.get_history()
+    expected_history = ["unfiltered_df = df.copy()"]
+    assert history == expected_history
+
+
+def test_apply_history():
+    df = create_df()
+    another_df = df.copy()
```
Comment (richardlin047) [inline modin_spreadsheet/tests/test_grid.py:883]: Rename to df_copy

File: modin_spreadsheet/tests/test_grid.py [hunk 7]
```diff
@@ -753,3 +753,199 @@ def test_edit_cell():
...
+            "filter_info": {
+                "field": "Date",
+                "type": "date",
+                "min": 1356998400000,
+                "max": 1357171199999,
+            },
+        }
+    )
+    spreadsheet._handle_view_msg_helper(
+        {"type": "change_sort", "sort_field": "Date", "sort_ascending": True}
```
Comment (richardlin047) [inline modin_spreadsheet/tests/test_grid.py:927]: Keep consistent with spacing between functions.

**Review by richardlin047 — CHANGES_REQUESTED** [review]
File deletion looks good, just some minor edits.

**Review by richardlin047 — CHANGES_REQUESTED** [review]
Think you forgot to commit the update_table functional call in the edit_cell logic section. Please add `self._update_table(triggered_by="edit_cell", fire_data_change_event=True)` to line 1528 in grid.py.

**Review by richardlin047 — APPROVED** [review]
Verified that it resolves #16, resolves #18.

@devin-petersohn let me know if you would like to review or if I should merge.

**Review by devin-petersohn — COMMENTED** [review]
Is it possible to run `black` on the test code?

**Comment by richardlin047** [comment]
Reformatted with black, no changes to the test code were needed.

**Review by devin-petersohn — APPROVED** [review]
LGTM, thanks @valerieyip!

**Pull request merged**